### PR TITLE
Feature/join preview

### LINKED ISSUE #67: Preview Join Mutation
Image

### PR DESCRIPTION
**Pull request by bewakes**
Resolves #67 

## Changes

* Add json validator for preview
* Mutation  and tests for table join preview

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/core/mutations.py [hunk 1]
```diff
@@ -35,6 +37,16 @@
 )
 
 
+class ColumnType(graphene.ObjectType):
+    key = graphene.String()
+    label = graphene.String()
```
Comment (k9845) [inline apps/core/mutations.py:42]: We can use `KeyLabelType` here as it is already defined.

Comment (k9845) [reply apps/core/mutations.py:42]: Also, make the type strict here.

File: apps/core/mutations.py [hunk 2]
```diff
@@ -239,6 +251,58 @@ def mutate(table, root, info, id, data: Any):
...
+                conflicting_col_suffix=str(target_table.id),
+            )
+        else:
+            return DiveMutationMixin(
+                errors=[
+                    dict(
+                        field="nonFieldErrors",
+                        messages=(
+                            "This clause is not supported at the moment."
+                            " Please use single inner join with equality clause."
```
Comment (k9845) [inline apps/core/mutations.py:294]:
```suggestion
                            "Please use single inner join with equality clause."
```

File: apps/core/tests/test_mutations.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
 from apps.core.factories import DatasetFactory, TableFactory
 from dive.consts import JOIN_CLAUSE_OPERATIONS
 from dive.base_test import create_test_file
+from .test_tasks import get_dummy_source_data, get_dummy_target_data, _test_expectation
```
Comment (k9845) [inline apps/core/tests/test_mutations.py:23]:
```suggestion
from .test_tasks import (
    get_dummy_source_data,
    get_dummy_target_data,
     _test_expectation
)
```

File: apps/core/tests/test_mutations.py [hunk 4]
```diff
@@ -356,3 +357,64 @@ def test_table_join(self, table_join_delay_func):
...
+                ),
+            },
+        }
+
+        resp_data = self.query_check(
+            preview_mutation,
+            variables=variables,
+        )
+        data = resp_data["data"]["joinPreview"]["result"]
+        _test_expectation(
```
Comment (k9845) [inline apps/core/tests/test_mutations.py:413]: @bewakes Any idea if we can test if new columns are appended after joining?

File: apps/core/mutations.py (hunk 2 above)
Comment (bewakes) [reply apps/core/mutations.py:294]: Grammar rules: there is space after a period.

File: apps/core/tests/test_mutations.py (hunk 3 above)
Comment (bewakes) [reply apps/core/tests/test_mutations.py:23]: Let black handle this.

File: apps/core/tests/test_mutations.py (hunk 4 above)
Comment (bewakes) [reply apps/core/tests/test_mutations.py:413]: It has been included.

File: apps/core/mutations.py (hunk 1 above)
Comment (bewakes) [reply apps/core/mutations.py]: Done

**Review by k9845 — APPROVED** [review]

**Pull request merged**